include/zephyr/drivers/sensor.h

```diff
@@ -45,6 +58,8 @@ struct sensor_value {
 	int32_t val1;
 	/** Fractional part of the value (in one-millionth parts). */
 	int32_t val2;
+	/** buffer part of the value*/
+	struct sensor_value_ex *ex;
```

Comment:
why pointer !

Reply:
during our last discussion you have give me this suggestion, I had the same doubt and asked the same here ⁠ https://discord.com/channels/938656347837309009/1022076962732261407/1091236252554104852